File: Backend/Todo-Service/Todo-Service/Dtos/TodoDto.cs
using MongoDB.Bson;
using System;

namespace Todo_Service.Dtos
{
	public class TodoDto
	{
		public string Id { get; set; }
		public string Name { get; set; }
		public string Status { get; set; }
		public DateTime? CreatedAt { get; set; }
		public string UserName { get; set; }
		public DateTime? LastUpdatedAt { get; set; }
		public DateTime? CompletedAt { get; set; } 
	}
}

File: Backend/todo-express-service/src/controllers/dtos/todo-dto.ts
import { ObjectID } from "mongodb";
import { Status } from "../../enum/status";
import { Todo } from "../../models/todo";

export class TodoDto {
    constructor(
        public id: string | ObjectID,
        public name: string,
        public status: Status,
        public createdAt: Date,
        public userName: string,
        public lastUpdatedAt: Date,
        public completedAt: Date
    ) { }

    public static fromTodo(todo: Todo) {
        return new TodoDto(todo._id,
            todo.name,
            todo.status,
            todo.createdAt,
            todo.userName,
            todo.lastUpdatedAt,
            todo.completedAt)
    }

}
